File: sample/src/main/java/com/your_company/ui_bindings_sample/CouponPlainActivity.java
package com.your_company.ui_bindings_sample;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

import com.nearit.ui_bindings.NearITUIBindings;
import com.your_company.ui_bindings_sample.factories.CouponFactory;

import it.near.sdk.reactions.couponplugin.model.Coupon;

/**
 * Created by Federico Boschini on 13/09/17.
 */

public class CouponPlainActivity extends AppCompatActivity {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_plain_coupon);

        CouponFactory couponFactory = new CouponFactory();

        //  In a real scenario the coupon is provided by the NearIT SDK
        Coupon validCoupon = couponFactory.getValidCoupon();

        Fragment couponFragment = NearITUIBindings.getInstance(this)
                .createCouponDetailFragmentBuilder(validCoupon)
                .disableAutoMaxBrightness()
                .build();

        getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, couponFragment).commit();

    }
}

File: sample-kotlin/src/main/java/com/your_company/ui_bindings_kotlin_sample/CouponPlainActivity.kt
package com.your_company.ui_bindings_kotlin_sample

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.nearit.ui_bindings.NearITUIBindings
import com.your_company.ui_bindings_kotlin_sample.factories.CouponFactory

/**
* @author Federico Boschini
*/

class CouponPlainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_plain_coupon)

        val coupon = CouponFactory.getValidCoupon()

        val couponFragment = NearITUIBindings.getInstance(this)
                .createCouponDetailFragmentBuilder(coupon)
                .build()

        supportFragmentManager
                .beginTransaction()
                .replace(R.id.fragment_container, couponFragment)
                .commit()

    }

}